presto-iceberg/src/main/java/com/facebook/presto/iceberg/IcebergPartition.java

```diff
@@ -0,0 +1,82 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.facebook.presto.iceberg;
+
+import com.fasterxml.jackson.annotation.JsonCreator;
+import com.fasterxml.jackson.annotation.JsonProperty;
+
+import javax.annotation.Nullable;
+
+import java.util.Objects;
+import java.util.Optional;
+
+import static com.google.common.base.MoreObjects.toStringHelper;
+import static java.util.Objects.requireNonNull;
+
+public final class IcebergPartition
```

Comment:
@agrawalreetika Why is IcebergPartition class needed? I see it adds a name field but that field is not used. HIveSplit has `List<HivePartitionKey> partitionKeys;`, and the `HivePartitionKey` is very similar to this IcebergPartition. So I assume you want to introduce a similar class to HivePartitionKey so we can serialize the split, is this right? If this is the case, 1) it should not be named IcebergPartition but IcebergPartitionKey. 2) the content is the same as HivePartitionKey. I think we can use use HivePartitionKey in IcebergSplit directly.

Reply:
@yingsu00 I have made the changes as discussed.